lib/processors/manifestCreator.js

```diff
@@ -525,6 +560,61 @@ async function createManifest(libraryResource, libBundle, descriptorVersion, _in
 		return sapUI5;
 	}
 
+	/**
+	 * Creates an i18n section:
+	 * - either using bundleUrl and supportedLocales
+	 * - or the i18n String
+	 *
+	 * @param {string} i18n bundle url, e.g. "messagebundle.properties"
+	 * @param {Map<string, Set<string>>} i18nToSupportedLocales cache to determine the supportedLocales only once
+	 * @returns {{bundleUrl: string, supportedLocales: string[]}|null|string} json structure with bundleUrl and
+	 *   supportedLocales or the i18n String if not a ".properties" file.
+	 *   <code>null</code> if given i18n String is <code>null</code>
+	 */
+	function createI18nSection(i18n, i18nToSupportedLocales) {
+		if (!i18n) {
+			return null;
+		}
+		if (!i18n.endsWith(".properties")) {
+			return i18n;
+		}
+
+		// if the supported locales information should not be included use i18n text
+		if (!includeSupportedLocalesInformation) {
+			return i18n;
+		}
+
+		let supportedLocales = i18nToSupportedLocales.get(i18n);
+
+		if (!supportedLocales) {
+			supportedLocales = new Set();
+
+			if (libBundle.findResource(i18n) != null) {
+				supportedLocales.add("");
+			}
+			const i18nPatternStr = i18n.substring(0, i18n.length - ".properties".length);
+			const i18nRegexp = new RegExp(i18nPatternStr + "_([^.]+)\\.properties$");
+
+
+			libBundle.getResources().forEach((resource) => {
+				const resPath = resource.getPath();
+				const result = i18nRegexp.exec(resPath);
+
+				if (result) {
+					supportedLocales.add(result[1].replace(/_/g, "-"));
+				}
+			});
+			i18nToSupportedLocales.set(i18n, supportedLocales);
+		}
+
+		const supportedLocalesArray = [...supportedLocales];
```

Comment:
While this is fine to convert the Set to an Array I think `Array.from(supportedLocales)` would be easier to understand.

Reply:
Interesting, I like the spread operator more, but that's a matter of taste. 

With both ways, I anyhow struggle with the details of iteration (whether it iterates entries - like for Object.entries or Map's default iterator - or only values).

Reply:
addressed with: c90b9a4